src/only-one/only-one.ts

```diff
@@ -27,8 +27,8 @@ const snake = new Snake()
 type AnimalInZoo = Monkey | Snake
 
 function eatSomething(animal: AnimalInZoo) {
-    animal.eatBanana();
-    animal.eatMouse();
+    if (animal instanceof Monkey) { return animal.eatBanana() };
+    if (animal instanceof Snake) { return animal.eatMouse() };
```

Comment:
Dobre rozwiązanie. Można też skrócić dla wygody [operatorem warunkowym](https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Operatory/Operator_warunkowy):
```suggestion
    return animal instanceof Monkey ? animal.eatBanana() : animal.eatMouse();
```